src/AlarmClock.h

```diff
@@ -56,87 +68,80 @@ template<typename Duration> class AlarmClock {
 
 protected:
 
-   void AlarmClockThread() {
-      SleepTimeIsBelow500ms() ? AlarmClock::SleepForFullAmount() : AlarmClock::SleepInIntervals();
+   void AlarmClockInterruptableThread() {
+      do {
+         // Call the sleep function
+         // cout << "THREAD: Calling sleep function" << endl;
+         mSleepFunction(kSleepTimeUsCount);
+         if (!mExit && !mReset) {
+            mExpired.store(true);
```

Comment:
please write a comment here why this code is here 
(i.e. due to unit test limitation)